src/windows/StringPolicy.cc

```diff
@@ -8,10 +8,31 @@
 using namespace Napi;
 
 StringPolicy::StringPolicy(const std::string name, const std::string &productName)
-    : RegistryPolicy(name, productName, REG_SZ) {}
+    : RegistryPolicy(name, productName, {REG_SZ, REG_MULTI_SZ}) {}
 
-std::string StringPolicy::parseRegistryValue(LPBYTE buffer, DWORD bufferSize) const
+std::string StringPolicy::parseRegistryValue(LPBYTE buffer, DWORD bufferSize, DWORD type) const
```

Comment:
The return type could be a `std::vector<std::wstring>` which will make the block slightly easier to process,

```c++
std::vector<std::wstring> StringPolicy::parseRegistryValue(LPBYTE buffer, DWORD bufferSize, DWORD type) const {
  std::vector<std::wstring> multi_string_values;
  std::vector<wchar_t> wideBuffer(bufferSize / sizeof(wchar_t));
  std::memcpy(wideBuffer.data(), buffer, bufferSize);
  auto current = wideBuffer.cbegin();
  auto end = wideBuffer.cend();
  while (current < end && *current != '\0') {
      auto current_end = std::find(current, end, '\0');
      multi_string_values.emplace_back(current, current_end);
      current = current_end + 1;
   }

  return multi_string_values;
}
```

Reply:
Thanks for the feedback. I liked the existing implementation mainly because it explains clearly how each type is handled even though it is lengthy. Mixing both could be bit confusing unless there is a perf gain.